### PR TITLE
PostChat Survey Link Mode Support and Storybook name and folder path update

### PR DESCRIPTION
**Pull request by abhinab22mohanty**
### Changes Introduced with PR

1. Added PostChat Survey Link mode support
2. Changed storybook publishing folder to docs/storybook
3. Updated names and view of storybook

### CODE REVIEW

**Comment by abhinab22mohanty** [comment]
Verified with following workflows:

1. Normal WF (Without PreChat or PostChat)
2. With PreChat Enabled
3. With PostChat Enabled (Survey Mode)
4. With PostChat Enabled (Link Mode)
5. OutOfOfficeHours WF

File: chat-widget/src/components/confirmationpanestateful/interfaces/IConfirmationPaneStatefulParams.ts [hunk 1]
```diff
@@ -10,5 +10,5 @@ export interface IConfirmationPaneStatefulParams extends IConfirmationPaneStatef
      * endChat: Internal Prop injected for triggering end of a chat using chatSDK
      */
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    endChat: (adapter: any) => Promise<void>;
+    endChat: (adapter: any, skipEndChatSDK?: boolean, skipCloseChat?: boolean) => Promise<void>;
```
Comment (charliewang95) [inline chat-widget/src/components/confirmationpanestateful/interfaces/IConfirmationPaneStatefulParams.ts:13]: comment for the parameters?

File: chat-widget/src/components/headerstateful/HeaderStateful.tsx [hunk 2]
```diff
@@ -24,7 +24,7 @@ export const HeaderStateful = (props: IHeaderStatefulParams) => {
     const [outOfOperatingHours, setOutOfOperatingHours] = useState(state.domainStates.liveChatConfig?.LiveWSAndLiveChatEngJoin?.OutOfOperatingHours === "True");
 
     const outOfOfficeStyleProps: IHeaderStyleProps = Object.assign({}, defaultOutOfOfficeHeaderStyleProps, outOfOfficeHeaderProps?.styleProps);
-
+    const conversationState = useRef(state.appStates.conversationState);
```
Comment (charliewang95) [inline chat-widget/src/components/headerstateful/HeaderStateful.tsx:27]: why is this needed?

File: chat-widget/src/components/headerstateful/interfaces/IHeaderStatefulParams.ts [hunk 3]
```diff
@@ -17,5 +17,5 @@ export interface IHeaderStatefulParams {
      * endChat: Internal Prop injected for triggering end of a chat using chatSDK
      */
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    endChat: (adapter: any) => Promise<void>;
+    endChat: (adapter: any, skipEndChatSDK?: boolean, skipCloseChat?: boolean) => Promise<void>;
```
Comment (charliewang95) [inline chat-widget/src/components/headerstateful/interfaces/IHeaderStatefulParams.ts:20]: comment here too

File: chat-widget/src/components/livechatwidget/common/endChat.ts [hunk 4]
```diff
@@ -13,31 +13,44 @@ import { defaultWebChatContainerStatefulProps } from "../../webchatcontainerstat
...
+            });
+            await chatSDK?.endChat();
+        } catch (ex) {
+            TelemetryHelper.logSDKEvent(LogLevel.ERROR, {
+                Event: TelemetryEvent.EndChatSDKCallFailed,
+                ExceptionDetails: {
+                    exception: ex
+                }
+            });
+            NotificationHandler.notifyError(NotificationScenarios.Connection, "End Chat Call Failed: " + ex);
```
Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/common/endChat.ts:30]: Remove this line. Customer doesn't need to know

File: chat-widget/stories/OmniChannelLiveChatWidget.stories.tsx [hunk 5]
```diff
@@ -18,9 +18,9 @@ const LiveChatWidgetTemplate: Story<ILiveChatWidgetProps> = (args) => <LiveChatW
     Live Chat Widget Default
 */
 
-export const LiveChatWidgetFixedSize = LiveChatWidgetTemplate.bind({});
+export const Default = LiveChatWidgetTemplate.bind({});
 
-const liveChatWidgetFixedSizeProps: ILiveChatWidgetProps = {
+const liveChatWidgetdefaultProps: ILiveChatWidgetProps = {
```
Comment (charliewang95) [inline chat-widget/stories/OmniChannelLiveChatWidget.stories.tsx:23]: liveChatWidget[D]efaultProps

File: chat-widget/src/components/headerstateful/HeaderStateful.tsx (hunk 2 above)
Comment (abhinab22mohanty) [reply chat-widget/src/components/headerstateful/HeaderStateful.tsx:27]: This is added to immediately update the controlProps in Header once there is a change with the app conversationState without triggering a re-render of the component

File: chat-widget/src/components/livechatwidget/common/initWebChatComposer.ts [hunk 6]
```diff
@@ -52,11 +53,15 @@ export const initWebChatComposer = (props: ILiveChatWidgetProps, chatSDK: any, s
             if (props?.webChatContainerProps?.renderingMiddlewareProps?.hideSendboxOnConversationEnd !== false) {
                 setWebChatStyles((styles: StyleOptions) => { return { ...styles, hideSendBox: true }; });
             }
-            if (isPostChatEnabled === "true" && postChatSurveyMode === PostChatSurveyMode.Embed) {
-                const loadPostChatEvent: ICustomEvent = {
-                    eventName: "LoadPostChatSurvey",
-                };
-                BroadcastService.postMessage(loadPostChatEvent);
+            if (isPostChatEnabled === "true") {
```
Comment (charliewang95) [inline chat-widget/src/components/livechatwidget/common/initWebChatComposer.ts:56]: Please update CHANGE_LOG with changes in this PR

**Review by charliewang95 — APPROVED** [review]
